subsys/bluetooth/mesh/rpl.c

```diff
@@ -121,8 +64,14 @@ bool bt_mesh_rpl_check(struct bt_mesh_net_rx *rx,
 		return false;
 	}
 
-	for (i = 0; i < ARRAY_SIZE(replay_list); i++) {
-		struct bt_mesh_rpl *rpl = &replay_list[i];
+	struct bt_mesh_rpl *rpl = NULL;
+
+	for (int i = 0; i < CONFIG_BT_MESH_CRPL; i++) {
+		rpl = bt_mesh_rpl_strg_iter(rpl);
+		if (!rpl) {
+			BT_ERR("Unable to retrieve RPL entry!");
+			break;
```

Comment:
This is quite dangerous. If there is broken customer implementation of rpl storage or somebody is able to misconfigure it then replay attack frames might sneak.

Reply:
Ok, should we just drop the error check and assume that it just works then?

Reply:
Removing NULL checking is even worse IMHO. You will get guaranteed the NULL address access with garbage values in rpl entry fields.